src/lib/png/png.c

```diff
@@ -0,0 +1,1129 @@
+		} else if(memcmp(chunk_name, "sRGB", 4) == 0) {
+			chunks_sum++;
+		} else if(memcmp(chunk_name, "sTER", 4) == 0) {
+			chunks_sum++;
+		} else if(memcmp(chunk_name, "tEXt", 4) == 0) {
+			chunks_sum++;
+		} else if(memcmp(chunk_name, "tIME", 4) == 0) {
+			chunks_sum++;
+		} else if(memcmp(chunk_name, "tRNS", 4) == 0) {
+			chunks_sum++;
+		} else if(memcmp(chunk_name, "zTXt", 4) == 0) {
+			chunks_sum++;
+		} else {
+			printf("Error : invalid chunk\n");
+			return -1;
+		}
+		pos += 12 + convert(chunk_length, 4);
+		lseek(fd, pos, SEEK_SET);
+	}
+	return chunks_sum;
+}
+
+/* Read IHDR chunk and get info */
+int read_IHDR(int fd, int *w, int *h, int *bd, int *color,
+											int *interlace) {
+	unsigned char chunk_length[4]; 		/* length of IDHR data (== 13 bytes) */
+	unsigned char chunk_name[4];   		/* ==IDHR */
+	unsigned char img_width[4], img_height[4];
+	unsigned char bit_depth; 			/* bit per sample */
+	unsigned char color_type;			/* 1 - pallete, 2 - truecolor, 4 - 2 + alpha */
+	unsigned char compression_type;		/* == 0 - deflate */
+	unsigned char filter_type;			/* == 0 - adaptive */
+	unsigned char interlace_method;		/* 0 - no interflace, 1 - Adam7interflace */
+	unsigned char crc[4]; 	       		/* CRC-32 */
+
+	lseek(fd, 8, SEEK_SET);
+
+	read(fd, chunk_length, 4);
+	if(convert(chunk_length, 4) != 13) {
+		printf("Error: length of IHDR != 13\n");
+		return -1;
+	}
+
+	read(fd, chunk_name, 4);
+	if(memcmp(chunk_name,"IHDR" ,4) != 0) {
+		printf("Error: invalid name of IHDR chunk\n");
+		return -1;
+	}
+
+	read(fd, img_width, 4);
+	*w = convert(img_width, 4);
+
+	read(fd, img_height, 4);
+	*h = convert(img_height, 4);
+
+	read(fd, &bit_depth, 1);
+	*bd = (int)bit_depth;
+
+	read(fd, &color_type, 1);
+	switch(color_type) {
+		case 1:
+			*color = PLT;
+			break;
+		case 2:
+			*color = TRCL;
+			break;
+		case 6:
+			*color = TRCLA;
+			break;
+		default:
+			printf("Error : invalid colortype\n");
+			return -1;
+	}
+
+	read(fd, &compression_type, 1);
+	if(compression_type != 0) {
+		printf("Error : invalid compression\n");
+		return -1;
+	}
+
+	read(fd, &filter_type, 1);
+	if(filter_type != 0) {
+		printf("Error : invalid filter\n");
+		return -1;
+	}
+
+	read(fd, &interlace_method, 1);
+	switch(interlace_method) {
+		case 0:
+			*interlace = 0;
+			break;
+		case 1:
+			*interlace = 1;
+			break;
+		default:
+			printf("Error : invalid interlace\n");
+			return -1;
+	}
+
+	read(fd, crc, 4);
+	return 0;
+}
+
+int check_png(int fd) {
+	unsigned char png_signature[8], fd_header[8];
+	/* first 8 bytes of png file*/
+	png_signature[0] = 0x89;
+	png_signature[1] = 0x50;
+	png_signature[2] = 0x4e;
+	png_signature[3] = 0x47;
+	png_signature[4] = 0x0d;
+	png_signature[5] = 0x0a;
+	png_signature[6] = 0x1a;
+	png_signature[7] = 0x0a;
+
+	lseek(fd, 0 , SEEK_SET);
+	read(fd, fd_header, 8);
+	if (!memcmp(png_signature, fd_header, 8)) {
+		return 0;
+	}
+	return -1;
+}
+
+void read_chunk(int fd, unsigned char *chunk_name) {
+	return;
```

Comment:
Empty function?

Reply:
we discussed that at this stage it is enough to read the main chunks, so now all additional chunks are processed with a stub